File: connectx-backend/src/main/java/com/connectx/message/dto/PagedMessageResponseDto.java
package com.connectx.message.dto;

import java.util.List;

public class PagedMessageResponseDto {

    private List<MessageDto> messages;
    private boolean hasMore;
    private Long nextCursor;
    private Integer limit;

    public PagedMessageResponseDto() {}

    public PagedMessageResponseDto(List<MessageDto> messages, boolean hasMore, Long nextCursor, Integer limit) {
        this.messages = messages;
        this.hasMore = hasMore;
        this.nextCursor = nextCursor;
        this.limit = limit;
    }

    public List<MessageDto> getMessages() {
        return messages;
    }

    public void setMessages(List<MessageDto> messages) {
        this.messages = messages;
    }

    public boolean isHasMore() {
        return hasMore;
    }

    public void setHasMore(boolean hasMore) {
        this.hasMore = hasMore;
    }

    public Long getNextCursor() {
        return nextCursor;
    }

    public void setNextCursor(Long nextCursor) {
        this.nextCursor = nextCursor;
    }

    public Integer getLimit() {
        return limit;
    }

    public void setLimit(Integer limit) {
        this.limit = limit;
    }
}

File: connectx-android/app/src/main/java/com/connectx/app/data/remote/message/model/PagedMessageResponseDto.kt
package com.connectx.app.data.remote.message.model

import kotlinx.serialization.Serializable

/**
 * Mirrors com.connectx.message.dto.PagedMessageResponseDto exactly -- the
 * backend's custom cursor-based pagination structure for message history
 * (GET /conversations/{id}/messages), NOT a Spring Page/Slice and NOT
 * offset/limit. `nextCursor` is the message ID to pass as the `before` query
 * parameter for the next page; `hasMore` indicates whether one exists.
 */
@Serializable
data class PagedMessageResponseDto(
    val messages: List<MessageDto> = emptyList(),
    val hasMore: Boolean = false,
    val nextCursor: Long? = null,
    val limit: Int? = null
)
